tenant/storage_user.go

```diff
@@ -275,8 +275,25 @@ func (s *Store) DeleteUser(ctx context.Context, tx kv.Tx, id influxdb.ID) error
 	if err != nil {
 		return UnavailablePasswordServiceError(err)
 	}
+	if err := ub.Delete(encodedID); err != nil {
+		return err
+	}
 
-	return ub.Delete(encodedID)
+	// Clean up user URMs.
+	urms, err := s.ListURMs(ctx, tx, influxdb.UserResourceMappingFilter{UserID: id})
+	if err != nil {
+		return err
+	}
+	// Do not fail fast on error.
+	// Try to avoid as much as possible the effects of partial deletion.
+	var aggErr error
+	for _, urm := range urms {
+		err := s.DeleteURM(ctx, tx, urm.ResourceID, urm.UserID)
+		if aggErr == nil {
+			aggErr = err
+		}
+	}
+	return aggErr
```

Comment:
is this the proper way of doing this? 🤔

Reply:
Your `aggErr` is actually just the "first error that occured". If that is what you want that works. If you want to display all the errors You could create a `type aggErr []error`. Then write an `Error()` function that concat's all the error messages from the errors in the list.